lib/bolt/transport/ssh.rb

```diff
@@ -167,8 +168,9 @@ def run_task(target, task, arguments, options = {})
               end
 
               if conn.run_as && stdin
-                wrapper = make_wrapper_stringio(remote_task_path, stdin)
+                wrapper = make_wrapper_stringio(remote_task_path, stdin, execute_options[:interpreter])
                 remote_wrapper_path = conn.write_remote_executable(dir, wrapper, 'wrapper.sh')
+                execute_options.delete(:interpreter)
```

Comment:
A comment would be helpful here, explaining that we don't need to use the interpreter to run the command since we've put it inside the script.